src/intrusive.rs

```diff
@@ -1,20 +1,31 @@
-//! libstd-based implementation of `event-listener`.
+//! Intrusive linked list-based implementation of `event-listener`.
 //!
-//! This implementation crates an intrusive linked list of listeners.
+//! This implementation crates an intrusive linked list of listeners. This list
+//! is secured using either a libstd mutex or a critical section.
 
 use crate::notify::{GenericNotify, Internal, Notification};
 use crate::sync::atomic::Ordering;
 use crate::sync::cell::{Cell, UnsafeCell};
-use crate::sync::{Mutex, MutexGuard};
 use crate::{RegisterResult, State, TaskRef};
 
+#[cfg(feature = "critical-section")]
+use core::cell::RefCell;
+#[cfg(all(feature = "std", not(feature = "critical-section")))]
+use core::ops::{Deref, DerefMut};
+
 use core::marker::PhantomPinned;
 use core::mem;
-use core::ops::{Deref, DerefMut};
 use core::pin::Pin;
 use core::ptr::NonNull;
 
-pub(super) struct List<T>(Mutex<Inner<T>>);
+pub(super) struct List<T>(
+    /// libstd-based implementation uses a normal Muetx to secure the data.
+    #[cfg(all(feature = "std", not(feature = "critical-section")))]
+    crate::sync::Mutex<Inner<T>>,
+    /// CS-based implementation uses a CS cell that wraps a RefCell.
```

Comment:
I think this should spell out `critical section` instead of `CS` at least once, otherwise it might lead to confusion.